crates/runtime/src/model/wrapper.rs

```diff
@@ -374,3 +431,127 @@ impl<S> Drop for TracedChatCompletionStream<S> {
         }
     }
 }
+
+fn template_system_prompt(
+    prompt: &str,
+    inputs: &HashMap<String, serde_json::Value>,
+) -> tera::Result<String> {
+    let mut t = Tera::default();
+    t.add_raw_template("system_prompt", prompt)?;
+    let mut context = tera::Context::new();
+    for (k, v) in inputs {
+        context.insert(k, v);
+    }
+    t.render("system_prompt", &context)
+}
+
+/// Return true if the system prompt is a template that would have its variables replaced with the input values.
+fn system_prompt_is_template_with_variables(prompt: &str) -> bool {
+    let mut t = Tera::default();
+    if t.add_raw_template("system_prompt", prompt).is_err() {
+        return false;
+    };
+
+    let Ok(tt) = t.get_template("system_prompt") else {
+        return false;
+    };
+    let result = has_variables_in_ast(&tt.ast);
+    tracing::debug!("Early check if system prompt='{prompt:?}' is template has result='{result}'. Not gating prompts.");
```

Comment:
So we're not actually returning if the system prompt has variables? The function name is misleading then.